src/model/Lotto.js

```diff
@@ -0,0 +1,51 @@
+import { checkNumber, checkNumberRange } from '../utils/validation.js';
+import { ERROR_MESSAGES } from '../constant/constants.js';
+import { createError } from '../utils/error.js';
+
+class Lotto {
+  #numbers;
+
+  constructor(numbers) {
+    this.#validate(numbers);
+    this.#numbers = numbers;
+  }
+
+  #validate(numbers) {
+    this.#checkNumbers(numbers);
+
+    if (numbers.length !== 6) {
+      throw new Error('[ERROR] 로또 번호는 6개여야 합니다.');
+    }
+
+    this.#checkNumbersRange(numbers);
+    this.#hasDuplicateNumbers(numbers);
+  }
+
+  #checkNumbers(numbers) {
+    for (let i = 0; i < numbers.length; i++) {
+      checkNumber(numbers[i], ERROR_MESSAGES.LOTTO.NOT_A_NUMBER);
+    }
+  }
+
+  #checkNumbersRange = (numbers) => {
+    for (let i = 0; i < numbers.length; i++) {
+      checkNumberRange(numbers[i], ERROR_MESSAGES.LOTTO.OUT_OF_RANGE);
+    }
+  };
```

Comment:
이 부분은 고차 함수로 작성해도 좋을 것 같아요!
```suggestion
#checkNumbers(numbers) {
  numbers.forEach(number => checkNumber(number, ERROR_MESSAGES.LOTTO.NOT_A_NUMBER));
}

#checkNumbersRange(numbers) {
  numbers.forEach(number => checkNumberRange(number, ERROR_MESSAGES.LOTTO.OUT_OF_RANGE));
}
```
이러면 가독성도 높이고 의미를 정확히 전달할 수 있어요